index.js

```diff
@@ -320,23 +320,30 @@ Router.prototype.off = function off (method, path) {
     const pathFull = path.replace(OPTIONAL_PARAM_REGEXP, '$1$2')
     const pathOptional = path.replace(OPTIONAL_PARAM_REGEXP, '$2')
 
-    this.off(method, pathFull)
-    this.off(method, pathOptional)
+    this.off(method, pathFull, derivedConstraints)
+    this.off(method, pathOptional, derivedConstraints)
     return
   }
 
+  function matcher (currentConstraints = {}) {
+    if (!derivedConstraints) return true
+
+    return Object.keys(derivedConstraints)
+      .every((key) => derivedConstraints[key] === currentConstraints[key])
```

Comment:
You should use deepEqual like here to compare constraints.
https://github.com/delvedor/find-my-way/blob/b38003a06b59da0ba336522babb933d465db3b96/custom_node.js#L207